frites/conn/conn_fit.py

```diff
@@ -0,0 +1,254 @@
+    mi_xy_t_pres = mi_xy_t[..., t_start]
+
+    # I(source_past; target_pres)
+    mi_x_st_pres = mi_x_sptf[..., t_start]
+
+    # I(target_past; target_pres) = mi_x_t
+    mi_x_t_pres = mi_x_tptf[..., t_start]
+
+    # Loop over delays for past of target and sources
+    for n_d in range(n_delays):
+
+        # Delay indices
+        delays = list(range(n_d, n_times - n_delays + n_d))
+
+        # PID with cue as target var
+        # I(target_{past}; cue)
+        mi_xy_t_past = mi_xy_t[..., delays]
+        # I(source_{past}; cue)
+        mi_xy_s_past = mi_xy_s[..., delays]
+
+        # redundancy between sources and target about S (MMI-based)
+        red_s_t = np.minimum(mi_xy_s_past, mi_xy_t_pres)
+        # redundancy between sources, target present and target past about S
+        red_all = np.minimum(red_s_t, mi_xy_t_past)
+        # first term of FIT with the cue as target var
+        fit_cue = red_s_t - red_all
+
+        # PID with target pres as target var
+        # redundancy between sources and target about target pres (MMI-based)
+        red_x_t = np.minimum(mi_xy_t_pres, mi_x_st_pres[:, n_d, :])
+        # redundancy between sources, target present and target past about S
+        red_all = np.minimum(red_x_t, mi_x_t_pres[:, n_d, :])
+        # second term of FIT with x pres as target var
+        fit_t_pres = red_x_t - red_all
+
+        if avg_delay:
+            fit[:, n_d, :] = np.minimum(fit_cue, fit_t_pres)
+        else:
+            fit += np.minimum(fit_cue, fit_t_pres)
+
+    # ________________________________ OUTPUTS ________________________________
+    # rebuild time vector
+    times = times[n_delays:]
+    delay = np.arange(1, n_delays + 1)[::-1] / cfg['sfreq']
+
+    # net transfer
+    if net:
+        roi_po = roi_p.copy()
+        done, roi_p, i_st = [], [], []
+        for n_s, (s, t) in enumerate(zip(i_s, i_t)):
+            # ignore if indices have already been stored
+            if ([s, t] in done) or ([t, s] in done): continue  # noqa
+            # find [source, target] and [target, source]
+            result = np.where((i_s == t) & (i_t == s))[0]
+            assert len(result) == 1
+            n_t = result[0]
+            # decide if (x->y - y->x) or (y->x - x->y)
+            r_s, r_t = roi_po[n_s], roi_po[n_t]
+            if not np.all(np.array([r_s, r_t]) == np.sort([r_s, r_t])):
+                n_s, n_t = n_t, n_s
+            r_s, r_t = roi_po[n_s].split('->')[0], roi_po[n_t].split('->')[0]
+            # store results
+            i_st.append([n_s, n_t])
+            roi_p.append(f"{r_s}-{r_t}")
+            done.append([s, t])
+        # computes net transfer
+        i_s, i_t = np.array(i_st).T
+        fit = fit[i_s, :] - fit[i_t, :]
+
+    # xarray conversion
+    if avg_delay:
+        dims, coords = ('roi', 'delays', 'times'), (roi_p, delay, times)
+    else:
+        dims, coords = ('roi', 'times'), (roi_p, times)
+    attrs['max_delay'] = max_delay
+    attrs['mi_type'] = mi_type
```

Comment:
Here can you add (those are the attributes of the `conn_fit` function. When saving the results, those attributes are conserved which is nice to share self-contained files with the configuration used):

```python
attrs['net'] = net
attrs['avg_delay'] = avg_delay
attrs['unit'] = 'Bits'
```